tests/ragger/test_blind_sign.py

```diff
@@ -72,65 +145,56 @@ def test_sign_parameter_selector(firmware: Firmware,
+    settings_toggle(firmware, navigator, [SettingID.DEBUG_DATA])
 
+    tx_params = common_tx_params()
+    data_len = len(bytes.fromhex(tx_params["data"][2:]))
+    # selector
+    flows = 1
+    data_len -= 4
+    # parameters
+    flows += data_len // 32
     with app_client.sign(BIP32_PATH, tx_params):
+        moves = []
         if firmware.device.startswith("nano"):
-            end_text = "Approve"
-            nav_inst = NavInsID.RIGHT_CLICK
-            valid_instr = [NavInsID.BOTH_CLICK]
-        else:
-            end_text = "Confirm"
-            nav_inst = NavInsID.USE_CASE_REVIEW_TAP
-            valid_instr = [NavInsID.USE_CASE_REVIEW_CONFIRM]
-
-        # Loop for "Selector" + the contract inputs
-        for step in range(count + 1):
-            navigator.navigate_until_text_and_compare(nav_inst,
-                                                      valid_instr,
-                                                      end_text,
-                                                      default_screenshot_path,
-                                                      f"{test_name}/step_{step}",
-                                                      screen_change_after_last_instruction=False)
-            step +=1
-
-        # Transaction review
-        if firmware.device.startswith("nano"):
-            end_text = "Accept"
+            if firmware.device == "nanos":
+                moves += [NavInsID.RIGHT_CLICK] * 2 + [NavInsID.BOTH_CLICK]
+                # Parameters on Nano S are split on multiple pages, hardcoded because the two parameters don't use the
+                # same amount of pages because of non-monospace fonts
+                moves += [NavInsID.RIGHT_CLICK] * 4 + [NavInsID.BOTH_CLICK]
+                moves += [NavInsID.RIGHT_CLICK] * 3 + [NavInsID.BOTH_CLICK]
+            else:
+                moves += ([NavInsID.RIGHT_CLICK] * 2 + [NavInsID.BOTH_CLICK]) * flows
+
+            if firmware.device == "nanos":
+                moves += [NavInsID.RIGHT_CLICK] * 2
+            else:
+                moves += [NavInsID.RIGHT_CLICK] * 4
+            moves += [NavInsID.BOTH_CLICK]
+
+            if firmware.device == "nanos":
+                pass
+                moves += [NavInsID.RIGHT_CLICK] * 9
+            else:
+                moves += [NavInsID.RIGHT_CLICK] * 5
+            moves += [NavInsID.BOTH_CLICK]
         else:
-            end_text = "Sign"
-        scenario_navigator.review_approve(default_screenshot_path, f"{test_name}/step_{step}", end_text)
+            moves += ([NavInsID.USE_CASE_REVIEW_TAP] * 2 + [NavInsID.USE_CASE_REVIEW_CONFIRM]) * flows
+            moves += [NavInsID.USE_CASE_CHOICE_REJECT]
+            moves += [NavInsID.USE_CASE_CHOICE_CONFIRM]
+            moves += [NavInsID.USE_CASE_REVIEW_TAP] * 3
+            moves += [NavInsID.USE_CASE_REVIEW_CONFIRM]
+        navigator.navigate_and_compare(default_screenshot_path,
```

Comment:
Why did you replace the `scenario_navigator`?
If we cannot use it any more, it seems the input parameter `scenario_navigator` could be removed?